packages/authentication/src/__tests__/acceptance/strategy-resolver/strategy.resolver.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright IBM Corp. 2018,2019. All Rights Reserved.
+// Node module: loopback4-example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Getter, inject} from '@loopback/context';
+import {Provider, ValueOrPromise} from '@loopback/core';
+import {AuthenticationMetadata} from '../../../decorators/authenticate.decorator';
+import {
+  extensionPoint,
+  extensions,
+} from '../../../decorators/authentication-extension.decorators';
+import {AuthenticationBindings} from '../../../keys';
+import {AuthenticationStrategy} from '../../../types';
+
+@extensionPoint('authentication-strategy')
+export class StrategyResolverProvider
+  implements Provider<AuthenticationStrategy | undefined> {
+  constructor(
+    @inject(AuthenticationBindings.METADATA)
+    private metadata: AuthenticationMetadata,
+    @extensions() // Sugar for @inject.getter(filterByTag({extensionPoint: 'greeter'}))
+    private authenticationStrategies: Getter<AuthenticationStrategy[]>,
```

Comment:
The getter here only captures an array of our authentication strategies, therefore I think we need another extension which has a getter for passport-strategies, like

```ts
@extensionPoint('passport-strategy')
export class PassportStrategyResolverProvider
  implements Provider<Strategy | undefined> {
  constructor(
    @inject(AuthenticationBindings.METADATA)
    private metadata: AuthenticationMetadata,
    @extensions()
    private passportStrategies: Getter<Strategy[]>,
  ) {}
  value(): ValueOrPromise<Strategy | undefined> {
    if (!this.metadata) {
      return;
    }
    // make sure the endpoint is decorated with a passport based strategy
    const isPassportStrategy =
          this.metadata.options && this.metadata.options.isPassportStrategy;
    if (!isPassportStrategy) return;

    const name = this.metadata.strategy;

    return this.findPassportStrategy(name).then(function(strategy) {
      if (strategy) {
        // Please note we wrap it with the adapter to return a Loopback auth strategy
        return new StrategyAdapter(strategy);
      } else {
        throw new Error(`The strategy '${name}' is not available.`);
      }
    });
  }

  async findPassportStrategy(name: string) {
    const strategies = await this.passportStrategies();
    const matchingAuthStrategy = strategies.find(a => a.name === name);
    return matchingAuthStrategy;
  }
}
```

Then the `StrategyResolverProvider` invokes the `PassportStrategyResolverProvider` when it finds the endpoint is decorated with a passport strategy.

Reply:
You can find the prototype implementation in [My PR](https://github.com/strongloop/loopback-next/blob/53cfeebbaf2ebe2897e28f51432f6fb5d1a4e946/packages/authentication/src/__tests__/acceptance/basic-auth.acceptance.ts#L163-L214).
There are 2 provider classes and each should be decorated with `extensionPoint()`.

Reply:
I will take a look. My strategy resolver in my PR will not deal with passport strategies at the moment. This will be handled by your PR. You can rebase your PR branch off of my branch as discussed.